sw/device/lib/dif/dif_i2c.h

```diff
@@ -90,6 +90,24 @@ typedef struct dif_i2c_timing_config {
   uint32_t scl_period_nanos;
 } dif_i2c_timing_config_t;
 
+/**
+ * Configuration for the addressing behavior of the I2C, can be disabled or
+ * configured to look for multiple addresses by masking certian bits.
+ */
+typedef struct dif_i2c_id {
+  /**
+   * Mask the I2C transactions before checking for a match. Recieved Address &
+   * mask must equal the programmed address to activate I2C Device. If Address &
+   * ~mask != 0, this will not match any addresses
+   */
+  uint8_t mask;
```

Comment:
We only support 7-bit addressing, right? And just checking, but the mask here doesn't ask for any left-shifting funny business, does it?

(Going off spec and using the R/W bit as part of the address is a pet peeve of mine, haha, but some chips specify it that way.)

Reply:
It doesn't ask for, nor permit left shifting and 15bit support is not supported by the hardware block (I think there may be the capacity to implement it in SW). I couldn't recall a clear concise way to specify the mask is to be aligned with the LSB and padded with a zero in the MSB. I'm open to suggestions. Is "LSB-aligned" clear?